source/common/runtime/runtime_impl.cc

```diff
@@ -23,60 +23,78 @@ namespace Runtime {
 
 const size_t RandomGeneratorImpl::UUID_LENGTH = 36;
 
+void RandomGeneratorImpl::random_bytes(uint8_t* out, size_t out_len) {
```

Comment:
Do you have coverage for the body of the first `if` here? I'd think testing at the boundaries (1 bytes `out_len`, 1023, 1024) would be good.

Do we even need this special case?